aztec/src/main/kotlin/org/wordpress/aztec/spans/MarkSpan.kt

```diff
@@ -1,12 +1,47 @@
 package org.wordpress.aztec.spans
 
+import android.graphics.Color
 import android.text.TextPaint
 import android.text.style.CharacterStyle
 import org.wordpress.aztec.AztecAttributes
+import org.wordpress.aztec.source.CssStyleFormatter
 
-class MarkSpan(override var attributes: AztecAttributes = AztecAttributes()) : CharacterStyle(), IAztecInlineSpan {
+class MarkSpan : CharacterStyle, IAztecInlineSpan {
     override var TAG = "mark"
 
-    override fun updateDrawState(tp: TextPaint?) {
+    override var attributes: AztecAttributes = AztecAttributes()
+    var textColor: Int? = null
+
+    constructor(attributes: AztecAttributes = AztecAttributes()) : super() {
+        this.attributes = attributes
+
+        val color = CssStyleFormatter.getStyleAttribute(attributes,
+            CssStyleFormatter.CSS_COLOR_ATTRIBUTE)
+        if (color.isNotEmpty()) {
+            textColor = Color.parseColor(color)
+        }
+    }
+
+    constructor(attributes: AztecAttributes = AztecAttributes(), colorString: String?) : super() {
+        this.attributes = attributes
+
+        if (colorString != null) {
+            textColor = Color.parseColor(colorString)
+        }
+    }
+
+    override fun updateDrawState(tp: TextPaint) {
+        configureTextPaint(tp)
+    }
+
+    private fun configureTextPaint(tp: TextPaint) {
+        if (textColor != null) {
+            tp.color = textColor as Int
+        }
+    }
+
+    fun getTextColor(): String {
```

Comment:
If I understand this correctly we are only accessing the `String` value of the `textColor` property via this method from outside the `MarkSpan` class. If this is the case we could make the `textColor` private and rename it to avoid confusion (the method name matches the [generated java getter name](https://kotlinlang.org/docs/java-to-kotlin-interop.html#properties) but returns a different type)

Reply:
Ohh I see so for this case, it'd make sense to rename this public function to something like `getTextHexColor` or something like that since the returned value is different from the private val `textColorValue` right?

Reply:
Yep. Or we can keep the method as is and rename the `textColor` to `textColorValue`.